src/ng/taskTrackerFactory.js

```diff
@@ -0,0 +1,122 @@
+'use strict';
+
+/**
+ * ! This is a private undocumented service !
+ *
+ * @name $$taskTrackerFactory
+ * @description
+ * A function to create `TaskTracker` instances.
+ *
+ * A `TaskTracker` can keep track of pending tasks (grouped by type) and can notify interested
+ * parties when all pending tasks (or tasks of a specific type) have been completed.
+ *
+ * @param {$log} log - A logger instance (such as `$log`). Used to log error during callback
```

Comment:
Doesn't this mean that the type of the `log` parameter is `$log`? Is that what you meant?

Reply:
Yes :grin: (Note: This is internal, non-public docs.)